parser.ts

```diff
@@ -166,34 +166,28 @@ class Parser {
   }
 
   private peek(): Token {
-    // tracer.trace(`peek (${this.current.kind})`);
     return this.current;
   }
 
-  private peekPrev(): Token {
-    // tracer.trace(`prev (${this.previous.kind})`);
-    return this.previous;
-  }
-
   private consume(kind: TokenKind, message: string): Token {
     return tracer.spanSync('consume', () => {
       tracer.trace('kind', kind);
-      if (this.check(kind)) return this.advance();
+      if (this.check(kind)) return this.advance() as Token;
       tracer.trace('failure:', message);
       const token: Token = this.peek();
       this.syntaxError(token, message);
     });
   }
 
-  private syntaxError(token: Token, message: string): void {
+  private syntaxError(token: Token, message: string): never {
     return tracer.spanSync('syntaxError', () => {
       tracer.trace('kind', token.kind);
       tracer.trace('message', message);
       const exc = new SyntaxError(message, {
         filename: this.filename,
         row: token.row,
         isLine: this.isLine,
-        lineNo: this.lineNo,
+        lineNo: this.lineNo === null ? -1 : this.lineNo,
```

Comment:
I'm wondering if lineNo should be nullable on Locations?

Reply:
It should probably be nullable. But this is OK incidentally - it's primarily used in formatting, and the formatting logic will treat -1 as an invalid line number.

I can refactor this later, but honestly I feel like syntax errors have bigger problems (like the use of isLine itself)